packages/@aws-cdk/aws-ecs/lib/capacity-provider.ts

```diff
@@ -0,0 +1,262 @@
+ */
+export interface CapacityProviderBase {
+  /**
+   * The name of the CapacityProvider
+   *
+   * @default - physical ID of the resource
+   */
+  readonly capacityProviderName?: string;
+
+  /**
+   * Whether to enable the managed termination protection
+   *
+   * @default True
+   */
+  readonly managedTerminationProtection?: boolean;
+
+  /**
+   * Whether to enable the managed scaling. This value will be overrided to be True
+   * if the `managedTerminationProtection` is enabled.
+   *
+   * @default - True.
+   */
+  readonly managedScaling?: boolean;
+
+  /**
+   * The maximum number of container instances that Amazon ECS will scale in or scale out at one time
+   *
+   * @default 10000
+   */
+  readonly maximumScalingStepSize?: number;
+
+  /**
+   * The minimum number of container instances that Amazon ECS will scale in or scale out at one time.
+   *
+   * @default 1
+   */
+  readonly minimumScalingStepSize?: number;
+
+  /**
+   * The target capacity value for the capacity provider. The specified value must be greater than `0` and less
+   * than or equal to `100`. A value of `100` will result in the Amazon EC2 instances in your Auto Scaling group
+   * being completely utilized.
+   *
+   * @default 100;
+   */
+  readonly targetCapacity?: number;
+
+}
+
+/**
+ * Options for addCapacityProvider
+ */
+export interface CapacityProviderOpts extends CapacityProviderBase {
+  /**
+   * capacity options for the autoscaling group
+   */
+  readonly capacityOptions: AddCapacityOptions;
+}
+
+
+/**
+ * Construct properties for CapacityProvider
+ */
+export interface CapacityProviderProps extends CapacityProviderBase {
+  /**
+   * The AutoscalingGroup for the CapacityProvider
+   */
+  readonly autoscalingGroup: IAutoScalingGroup;
+
+}
+
+/**
+ * A CapacityProvider represents the Amazon ECS Capacity Provider
+ */
+export class CapacityProvider extends cdk.Resource implements ICapacityProvider {
+  /**
+   * Import an existing CapacityProvider into this CDK app.
+   */
+  public static fromCapacityProviderName(scope: cdk.Construct, id: string, capacityProviderName: string): ICapacityProvider {
+    class Import extends cdk.Resource implements ICapacityProvider {
+      public readonly capacityProviderName = capacityProviderName;
+    }
+    return new Import(scope, id);
+  }
+
+  /**
+   * The name of the CapacityProvider
+   */
+  public readonly capacityProviderName: string;
+  private _managedTerminationProtection: boolean
+  constructor(scope: cdk.Construct, id: string, props: CapacityProviderProps) {
+    super(scope, id);
+
+    this._managedTerminationProtection = props.managedTerminationProtection ?? true;
+
+    const onEvent = new lambda.Function(this, 'InstanceProtectionHandler',  {
+      runtime: lambda.Runtime.PYTHON_3_8,
+      handler: 'index.on_event',
+      timeout: cdk.Duration.seconds(60),
+      environment: {
+        autoscaling_group_name: props.autoscalingGroup.autoScalingGroupName,
```

Comment:
`ManagedTerminationProtection` is passed from here as a property of the custom resource:

https://github.com/aws/aws-cdk/blob/f60337c361582a71bd71abf630af5542d98ccdbd/packages/%40aws-cdk/aws-ecs/lib/capacity-provider.ts#L150

And the lambda function get the property from here

https://github.com/aws/aws-cdk/blob/f60337c361582a71bd71abf630af5542d98ccdbd/packages/%40aws-cdk/aws-ecs/lib/instance-protection-handler/index.py#L16

We also need to pass the ASG name to the custom resource and I believe we should pass the it as a property instead of a lambda env var. I'll try update this.